File: flink-jobs/src/main/java/ru/spb/itmo/pirsbd/asashina/flinkjobs/model/ClickEventPayload.java
package ru.spb.itmo.pirsbd.asashina.flinkjobs.model;

import java.util.Map;

public record ClickEventPayload(String eventTitle, String elementId, Integer x, Integer y, String elementText,
                                String elementClass, String pageTitle, Integer viewportWidth, Integer viewportHeight,
                                Double scrollPosition, Long timestampOffset, Map<String, Object> metadata) {

}

File: common-dto/src/main/kotlin/ru/spb/pirsbd/asashina/common/dto/ClickEventPayload.kt
package ru.spb.pirsbd.asashina.common.dto

data class ClickEventPayload(
    var eventTitle: String? = null,
    var elementId: String? = null,
    var x: Int? = null,
    var y: Int? = null,
    var elementText: String? = null,
    var elementClass: String? = null,
    var pageTitle: String? = null,
    var viewportWidth: Int? = null,
    var viewportHeight: Int? = null,
    var scrollPosition: Double? = null,
    var timestampOffset: Long? = null,
    var metadata: Map<String, Any>? = null
)